pkg/datapath/iptables/iptables.go

```diff
@@ -438,148 +455,167 @@ func (m *IptablesManager) installStaticProxyRules() error {
-			err = ip6tables.runProg([]string{
-				"-t", "raw",
-				"-A", ciliumOutputRawChain,
-				"-m", "mark", "--mark", matchProxyReply,
-				"-m", "comment", "--comment", "cilium: NOTRACK for proxy return traffic",
-				"-j", "CT", "--notrack"}, false)
-		}
-		if err == nil {
-			// Explicit ACCEPT for the proxy return traffic. Needed when the OUTPUT defaults to DROP.
-			// Matching needs to be the same as for the NOTRACK rule above.
-			err = ip6tables.runProg([]string{
-				"-t", "filter",
-				"-A", ciliumOutputChain,
-				"-m", "mark", "--mark", matchProxyReply,
-				"-m", "comment", "--comment", "cilium: ACCEPT for proxy return traffic",
-				"-j", "ACCEPT"}, false)
-		}
-		if err == nil && m.haveSocketMatch {
+			"-j", "CT", "--notrack"}); err != nil {
+			return err
+		}
+
+		// Explicit ACCEPT for the proxy traffic. Needed when the INPUT defaults to DROP.
+		// Matching needs to be the same as for the NOTRACK rule above.
+		if err := ip6tables.runProg([]string{
+			"-t", "filter",
+			"-A", ciliumInputChain,
+			"-m", "mark", "--mark", matchToProxy,
+			"-m", "comment", "--comment", "cilium: ACCEPT for proxy traffic",
+			"-j", "ACCEPT"}); err != nil {
+			return err
+		}
+
+		// No conntrack for proxy return traffic
+		if err := ip6tables.runProg([]string{
+			"-t", "raw",
+			"-A", ciliumOutputRawChain,
+			"-m", "mark", "--mark", matchProxyReply,
+			"-m", "comment", "--comment", "cilium: NOTRACK for proxy return traffic",
+			"-j", "CT", "--notrack"}); err != nil {
+			return err
+		}
+
+		// Explicit ACCEPT for the proxy return traffic. Needed when the OUTPUT defaults to DROP.
+		// Matching needs to be the same as for the NOTRACK rule above.
+		if err := ip6tables.runProg([]string{
+			"-t", "filter",
+			"-A", ciliumOutputChain,
+			"-m", "mark", "--mark", matchProxyReply,
+			"-m", "comment", "--comment", "cilium: ACCEPT for proxy return traffic",
+			"-j", "ACCEPT"}); err != nil {
+			return err
+		}
+
+		if m.haveSocketMatch {
 			// Direct inbound TPROXYed traffic towards the socket
-			err = ip6tables.runProg(m.inboundProxyRedirectRule("-A"), false)
+			if err := ip6tables.runProg(m.inboundProxyRedirectRule("-A")); err != nil {
+				return err
+			}
 		}
 	}
-	return err
+
+	return nil
 }
 
-func (m *IptablesManager) doCopyProxyRules(prog iptablesInterface, table string, re *regexp.Regexp, match, oldChain, newChain string) {
-	output, err := prog.runProgCombinedOutput([]string{"-t", table, "-S"}, true)
+func (m *IptablesManager) doCopyProxyRules(prog iptablesInterface, table string, re *regexp.Regexp, match, oldChain, newChain string) error {
+	rules, err := prog.runProgCombinedOutput([]string{"-t", table, "-S"})
 	if err != nil {
-		log.WithFields(logrus.Fields{
-			"table": table,
-			"prog":  prog.getProg(),
-		}).WithError(err).Warning("Cannot list rules in table. Layer 7 proxy port may get reallocated, which could cause disruption for traffic selected by L7 policy", prog.getProg(), table)
```

Comment:
I don't think the log is relevant anymore as now in case of failure we keep retrying instead of just ignoring the error (and we would also log an error message related to the failure of the list command)

Reply:
Ok, but if we retry X times and eventually fail for good, what's the error message we will show? Will it highlight the potential disruption of traffic selected by L7 policies? (:point_left: that's the information I fear is now missing for users.)

Is there any downside to passing a new error with that message and the `err` instead of just `err`?

Reply:
> Is there any downside to passing a new error with that message and the err instead of just err?

no downside, I just added additional context about L7 policies traffic disruption to the error returned by `copyProxyRule`